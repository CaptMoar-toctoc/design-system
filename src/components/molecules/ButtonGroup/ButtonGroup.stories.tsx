import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import ButtonGroup from './ButtonGroup';


export default {
    title: 'TocToc-DS/Molecules',
    component: ButtonGroup,
} as ComponentMeta<typeof ButtonGroup>;

const Template: ComponentStory<typeof ButtonGroup> = (args) => <ButtonGroup {...args} />;

export const ButtonGroupTT = Template.bind({});

ButtonGroupTT.args = {
    handlerFn: () => { console.log(`click`) },
};