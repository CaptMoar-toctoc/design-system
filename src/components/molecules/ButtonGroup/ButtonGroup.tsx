import * as React from 'react';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
import { Typography, styled, ThemeProvider } from '@mui/material';
import theme from '../../theme/theme';

const StyledButtonGroup = styled(ToggleButtonGroup)(() => ({
	'& .Mui-selected': {
		color: '#3200C1',
		borderWidth: '4px',
		borderStyle: 'solid',
		borderColor: '#3200C1',
		borderRadius: '2px'
	},
	'& .MuiButtonBase-root': {
		borderRight: 'none',
		borderLeft: 'none',
		borderTop: 'none',
		fontWeight: 600,
		lineHeight: '132%',
		padding: '7px',
		backgroundColor: '#ffffff',
	}
}));

export default function ButtonGroup() {
	const [alignment, setAlignment] = React.useState<string | null>('left');

	const handleAlignment = (
		event: React.MouseEvent<HTMLElement>,
		newAlignment: string | null,
	) => {
		setAlignment(newAlignment);
	};
	return (
		<ThemeProvider theme={theme}>
			<StyledButtonGroup
				value={alignment}
				exclusive
				onChange={handleAlignment}
				aria-label="listing of quoted properties"
			>
				<ToggleButton value="left" aria-label="onsale new project">
					<Typography>Proyectos nuevos en venta</Typography>
				</ToggleButton>
				<ToggleButton value="center" aria-label="onsale pre-owned">
					<Typography>Propiedades usadas en venta</Typography>
				</ToggleButton>
				<ToggleButton value="right" aria-label="renting pre-owned">
					<Typography>Propiedades usadas en arriendo</Typography>
				</ToggleButton>
			</StyledButtonGroup>
		</ThemeProvider>

	);
}