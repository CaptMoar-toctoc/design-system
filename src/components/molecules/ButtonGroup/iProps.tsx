import { ReactNode } from "react";
export interface iProps {
	text: Array<string>,
	children?: ReactNode,
	handlerFn?: any,
	disabled?: boolean,
	href: string,
	ariaLabel: string
}