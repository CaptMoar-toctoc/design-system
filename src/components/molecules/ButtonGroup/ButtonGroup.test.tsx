import ButtonGroup from './ButtonGroup';
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';


describe("ButtonGroup testing", () => {

    test("Render ButtonGroup component", () => {
        render(<ButtonGroup />)
    })

    test("Render SnapShot", () => {
        expect(render(<ButtonGroup />)).toMatchSnapshot();
    })

    test.todo("It has a rendered text"), () => {
        render(<ButtonGroup />);
        const text = screen.queryByText('Proyectos')
        expect(text).not.toBeNull();
    }

    test.todo("Is a clickable component"), () => {
        render(<ButtonGroup />)
        const buttonGroup = screen.getByRole('button')
        fireEvent.click(buttonGroup)
        expect(buttonGroup).toBeTruthy();
    }
})

export { }