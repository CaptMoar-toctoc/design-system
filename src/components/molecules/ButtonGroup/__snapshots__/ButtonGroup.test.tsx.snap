// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup testing Render SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="listing of quoted properties"
        class="MuiToggleButtonGroup-root css-1hnhqu0-MuiToggleButtonGroup-root"
        role="group"
      >
        <button
          aria-label="onsale new project"
          aria-pressed="true"
          class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-pfnqtg-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="left"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-10a3z0g-MuiTypography-root"
          >
            Proyectos nuevos en venta
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="onsale pre-owned"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-pfnqtg-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="center"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-10a3z0g-MuiTypography-root"
          >
            Propiedades usadas en venta
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="renting pre-owned"
          aria-pressed="false"
          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-pfnqtg-MuiButtonBase-root-MuiToggleButton-root"
          tabindex="0"
          type="button"
          value="right"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-10a3z0g-MuiTypography-root"
          >
            Propiedades usadas en arriendo
          </p>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="listing of quoted properties"
      class="MuiToggleButtonGroup-root css-1hnhqu0-MuiToggleButtonGroup-root"
      role="group"
    >
      <button
        aria-label="onsale new project"
        aria-pressed="true"
        class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-pfnqtg-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="left"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-10a3z0g-MuiTypography-root"
        >
          Proyectos nuevos en venta
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="onsale pre-owned"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-pfnqtg-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="center"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-10a3z0g-MuiTypography-root"
        >
          Propiedades usadas en venta
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="renting pre-owned"
        aria-pressed="false"
        class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-pfnqtg-MuiButtonBase-root-MuiToggleButton-root"
        tabindex="0"
        type="button"
        value="right"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-10a3z0g-MuiTypography-root"
        >
          Propiedades usadas en arriendo
        </p>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
